### PR TITLE
[Feature] : Work page

### PR DESCRIPTION
**Pull request by SahilVermaFED**
## Description

-   [Ticket Link]
-   Work Page Completed and includes :
     1. Scroll 0 is given so that if we move to any page and comes back to work page then it always starts from top after all pages complition we use this as a commons loader for all pages.
     2. Proptype is used so that it automatically gets what type of prop is it .
     3. Prop is used for changing NavLink for pages as  given.

## UI screenshot or
[Screencast from 19-03-24 11:08:40 AM IST.webm](https://github.com/primathonacademy/april-batch-live-project/assets/131689244/ee479f51-7ee2-4463-9c9c-6778234ea2b3)
 Video

### General:

-   [x] **PR Size :** Confirming PR size < 20 files or pre-approved if > 20 files.
-   [x] **Self Review :** Confirming I have self-reviewed my PR
-   [x] **Peers Review :** - [ ] @Dev 1
-   [x] Confirming I have added all sensitive information-related should be inside .env files

### Client:

-   [x] **Console :** Confirming no console errors in the UI for this update.

-   [x] **Import order :** Confirming correct import order.

```
# External Import
import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';

# Internal Import
import { } from '../../Api/ApiRoutes';
import { } from '../../constants/AppConstant';
```

-   [x] **Re-renders :** Confirming minimal re-renders for new React components.

-   [x] **Network calls :** Confirming efficient network/API calls, minimized as needed.

-   [x] **Util function comments :** Confirming comprehensive comments for utility functions.

-   [x] **Image optimization :** Confirming I have checked that all images are optimized. If not, please use https://tinyjpg.com/ or https://vecta.io/nano.

-   [x] **Dead code :** Confirming I have removed all comments code which is of no use OR written why I have kept the commented code.

### Confirming different viewport testing of the entire flow:

-   [x] Tested on 280px viewport
-   [x] Tested on 320px viewport
-   [x] Tested on 375px viewport
-   [x] Tested on 425px viewport
-   [x] Tested on 768px viewport
-   [x] Tested on 1024px viewport
-   [x] Tested on 1440px viewport
-   [x] Tested on 2560px viewport

### End-to-end testing:

-   [x] **Happy flow :** Successfully tested end-to-end happy flow, all working as expected.
-   [x] **Error flow :** Tested UI flow with error scenarios, gracefully handled errors on UI
-   [x] **Common component changed :** Tested change in common component across views to prevent breaking elsewhere.

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -14,6 +14,7 @@ module.exports = {
     plugins: ['react-refresh'],
     rules: {
         'react/jsx-no-target-blank': 'off',
-        'react-refresh/only-export-components': ['warn', { allowConstantExport: true }]
+        'react-refresh/only-export-components': ['warn', { allowConstantExport: true }],
+        "prettier/prettier": "off"
```
Comment (ram-primathon) [inline .eslintrc.cjs]: You can't do this you are not allow to change this file

**Review by ram-primathon — CHANGES_REQUESTED** [review]

File: .eslintrc.cjs (hunk 1 above)
Comment (TheSahilVerma) [reply .eslintrc.cjs]: i have done this by mistake sir i am resolving this right now

File: package-lock.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
             "version": "0.0.0",
             "dependencies": {
                 "@gsap/react": "^2.1.0",
+                "prop-types": "^15.8.1",
```
Comment (ram-primathon) [inline package-lock.json]: WHy we need this package?

Comment (TheSahilVerma) [reply package-lock.json]: This package includes runtime type checking for React props and similar objects.

Comment (TheSahilVerma) [reply package-lock.json]: done

File: src/components/WorkComponent.jsx [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
+import scrollHandler from '../utils/scrollHandlerUtils';
+import { workImg1, workImg2, workImg3, workImg4, rightArrow } from '../assets/index';
+import PropTypes from 'prop-types';
 
-const WorkComponent = () => {
+const WorkComponent = (props) => {
+    const { isHomePage } = props;
     const boxRefs = useRef([]);
 
+    WorkComponent.propTypes = {
```
Comment (ram-primathon) [inline src/components/WorkComponent.jsx]: this should be outside of component

**Review by ram-primathon — CHANGES_REQUESTED** [review]

**Review by ram-primathon — APPROVED** [review]

**Pull request merged**